fxa-auth-db-server/test/backend/db_tests.js

```diff
@@ -832,13 +839,26 @@ module.exports = function(config, DB) {
                 return devices[0]
               })
               .then(function(device) {
-                t.equal(device.name, newDevice.name, 'name')
+                t.deepEqual(device.sessionTokenId, SESSION_TOKEN_ID, 'sessionTokenId')
+                t.equal(device.name, deviceInfo.name, 'name')
                 t.deepEqual(device.id, deviceId, 'id')
-                t.equal(device.createdAt, newDevice.createdAt, 'createdAt')
-                t.equal(device.type, newDevice.type, 'type')
+                t.equal(device.createdAt, deviceInfo.createdAt, 'createdAt')
+                t.equal(device.type, deviceInfo.type, 'type')
+                t.equal(device.callbackURL, deviceInfo.callbackURL, 'callbackURL')
+                t.deepEqual(device.callbackPublicKey, deviceInfo.callbackPublicKey, 'callbackPublicKey')
                 t.ok(device.lastAccessTime > 0, 'has a lastAccessTime')
+                return db.createSessionToken(newSessionTokenId, SESSION_TOKEN)
+              })
+              .then(function () {
                 return db.upsertDevice(ACCOUNT.uid, deviceId, {
-                  name: 'updated name'
+                  sessionTokenId: newSessionTokenId,
+                  name: 'updated name',
+                  type: null,
+                  callbackURL: '',
+                  callbackPublicKey: zeroBuffer32
+                })
```

Comment:
IIRC, part of what this is testing is sensible handling of partial updates, so let's be careful we don't lose that by specifying all the extra device info here.